postgres/src/test/java/org/jdbi/v3/postgres/TestLobStream.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.postgres;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.sql.SQLException;
+import java.util.function.Supplier;
+
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+import org.jdbi.v3.testing.JdbiRule;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.postgresql.PGConnection;
+import org.postgresql.largeobject.LargeObjectManager;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestLobStream {
+    private static final int BIGLY = 1024 * 1024 * 64;
+
+    @ClassRule
+    public static JdbiRule db = PostgresDbRule.rule();
+
+    static Handle h;
+    static LargeObjectManager mgr;
+    static Lobject lob;
+
+    @BeforeClass
+    public static void setup() throws SQLException {
+        h = db.getHandle();
+        h.execute("CREATE TABLE lob (id int, lob oid)");
+        mgr = h.getConnection().unwrap(PGConnection.class).getLargeObjectAPI();
+        lob = h.attach(Lobject.class);
+    }
+
+    @Test
+    public void blobCrud() throws IOException {
+        final Supplier<InputStream> expectedData = () -> new IntsInputStream(0, 255);
+        h.useTransaction(th -> {
+            lob.insert(1, expectedData.get());
+            assertSameBytes(lob.findBlob(1), expectedData.get());
+            lob.deleteLob(1);
+            assertThat(lob.findBlob(1)).isNull();
+        });
+    }
+
+    @Test
+    public void clobCrud() throws IOException {
+        final Supplier<Reader> expectedData = () -> new IntsReader('0', 'z');
+        h.useTransaction(th -> {
+            lob.insert(2, expectedData.get());
+            assertSameChars(lob.findClob(2), expectedData.get());
+        });
+    }
+
+    private void assertSameBytes(InputStream a, InputStream b) throws IOException {
+        int pos = 0;
+        int read;
+        while ((read = a.read()) != -1) {
+            assertThat(b.read())
+                    .describedAs("byte at position %s", pos)
```

Comment:
is it a byte though? An int is 4 bytes

Reply:
I mean, the test limits the bounds to [0, 255], but everything about the impl says ints, so...

Reply:
Yeah, `InputStream` works on bytes but the type is `int` since its *unsigned* bytes.

Reply:
Sure, but from a functional perspective this impl reads like two different books. Might as well use the `byte` type and name everything as such

Reply:
You can't use `byte` because it's a signed byte -- `-128..127` -- `InputStream` requires you to work in `0..255` which cannot fit in a Java `byte`.  This isn't a Jdbi-ism, this is a `java.io`-ism that we can't change.

Reply:
Also, the return value has a special flag value `-1` for EOF which could not also fit in a byte